### PR TITLE
feat: intent-based PR picker and role-aware actions

### PR DESCRIPTION
**Pull request by donovan-yohan**
## Summary

Adds a unified intent-based picker modal and role-aware PR actions. Reviewers now see "Review" instead of "Merge" on open PRs, and a new picker (`/` or `Cmd+K`) resolves PRs, branches, and issues into contextual session actions — review, fix conflicts, start from issue, resume, etc.

## Changes

- **Bug fix:** Branch picker dropdown was clipped by `overflow:hidden` on PrTopBar
- **Role-aware state machine:** `derivePrAction` now accepts `role` — authors see Merge, reviewers see Review, CI status and draft state are properly detected
- **SessionIntent abstraction:** New `resolveIntent()` maps PRs/branches/issues into actionable intents based on role and existing sessions
- **Dashboard fix:** RepoDashboard, OrgDashboard, and App.svelte now pass real role, CI data, and isDraft to the state machine
- **OpenPicker component:** Tabbed modal (all/prs/branches/issues) with search, keyboard nav, and intent action buttons per row
- **App wiring:** `/` and `Cmd+K` open the picker; intent selection creates/resumes sessions or opens GitHub

## Testing

- 40 tests pass (31 pr-state + 9 session-intent)
- TypeScript compiles clean
- Frontend builds with 0 new errors
- Role-aware matrix verified: author/reviewer × all-clear/conflicts/CI-failing/draft/unresolved-comments

---
*Created with `/pr:author`*

### CODE REVIEW

**Review by donovan-yohan — COMMENTED** [review]
## PR Review Summary

**Agents run:** code-reviewer, silent-failure-hunter, pr-test-analyzer, type-design-analyzer, comment-analyzer, code-simplifier

**ADR compliance:** Compliant — no architecture rule violations.

### Overview

Clean architecture overall — the intent resolution system, state machine extension, and picker are well-designed. The `derivePrAction` state table is excellent. Svelte 5 runes usage, DESIGN.md compliance, and import conventions all look correct.

### Key issues by severity

| Severity | Count | Summary |
|----------|-------|---------|
| Critical/High | 4 | Duplicate worktree bug, silent failures in picker handlers, `mapPrActionToIntent` type safety, fire-and-forget DELETE |
| Warning | 7 | DRY violations (4x `buildPrStateInput`, 3x `colorToVariant`), handler duplication, O(n²) indexOf, missing test coverage |
| Info | 10+ | Comment accuracy, unused params, lossy heuristics, minor edge cases |

### Pre-existing code issues (not in diff, noted here)

- **App.svelte:619** — `[silent-failure-hunter]` Empty catch silently swallows all errors from `fetchWorkspaceSettings`. Fix: add `console.warn`.
- **App.svelte:815** — `[silent-failure-hunter]` Fire-and-forget DELETE with no `.catch()`. If it fails, `refreshAll()` never runs and session list becomes stale. Fix: add `.catch()`.

### Agents with no blocking issues
- **comment-analyzer**: 2 inaccurate comments flagged (JSDoc and grouping label), no blockers
- **code-simplifier**: DRY violations flagged but no correctness issues

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: frontend/src/App.svelte [hunk 1]
```diff
@@ -685,17 +705,85 @@
...
+      }
+      case 'archive': {
+        break;
+      }
+    }
+  }
+
+  async function handleOpenBranchSession(branchName: string, repoPath: string, prompt?: string) {
+    try {
+      const wt = await createWorktree(repoPath, branchName);
```
Comment (donovan-yohan) [inline frontend/src/App.svelte]: **[code-reviewer]** Bug: `handleOpenBranchSession` always calls `createWorktree` without checking for existing sessions or worktrees. Unlike `handleOpenPrBranch` (lines 663-706) which checks for `existingSession` and `existingWorktree`, this function skips that entirely.

`resolveIntent` already populates `intent.existingWorktreePath` (session-intent.ts:152), but `handlePickerIntent` never passes it through. This will error when selecting a branch that already has a worktree.

Fix: Add existence checks mirroring `handleOpenPrBranch`, and pass `intent.existingWorktreePath` from the call sites.

File: frontend/src/App.svelte [hunk 2]
```diff
@@ -685,17 +705,85 @@
...
+        break;
+      }
+      case 'merge-pr': {
+        if (item.kind === 'pr') {
+          window.open(item.pr.url, '_blank');
+        }
+        break;
+      }
+      case 'open-branch': {
+        if (item.kind === 'branch' && activeWorkspace) {
```
Comment (donovan-yohan) [inline frontend/src/App.svelte]: **[code-reviewer]** Bug: Uses `activeWorkspace.path` instead of `item.repoPath`. The branch `PickerItem` already carries `repoPath` (set in OpenPicker.svelte:112). Using `activeWorkspace.path` means:
1. If the user switches workspace while the picker is open, the branch opens in the wrong repo
2. The action silently does nothing if `activeWorkspace` is null

Fix: Use `item.repoPath` directly:
```typescript
case 'open-branch': {
  if (item.kind === 'branch') {
    await handleOpenBranchSession(item.name, item.repoPath, intent.prompt ?? undefined);
  }
  break;
}
```

File: frontend/src/lib/session-intent.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      existingSessionId: existingSession.id,
+    };
+    if (existingSession.worktreePath) resumeIntent.existingWorktreePath = existingSession.worktreePath;
+    intents.push(resumeIntent);
+  }
+
+  return intents;
+}
+
+function mapPrActionToIntent(actionType: string): SessionIntentType {
```
Comment (donovan-yohan) [inline frontend/src/lib/session-intent.ts]: **[type-design-analyzer]** `mapPrActionToIntent` accepts `string` instead of `PrActionType`, erasing type safety. The `default` fallback silently maps `'none'`, `'ready-for-review'`, and `'checks-running'` to `'open-branch'`.

When `derivePrAction` returns `type: 'none'` (reviewer on draft PR), this produces an intent with empty label and `'none'` color — rendering an empty button in the picker.

Fix: Type the param as `PrActionType`, handle all 11 variants explicitly, and add an exhaustiveness check:
```typescript
function mapPrActionToIntent(actionType: PrActionType): SessionIntentType {
  // ... explicit cases ...
  default: {
    const _exhaustive: never = actionType;
    return 'open-branch';
  }
}
```

File: frontend/src/App.svelte [hunk 4]
```diff
@@ -685,17 +705,85 @@
...
+        break;
+      }
+      case 'start-from-issue': {
+        if (item.kind === 'issue') {
+          const branchName = issueToBranchName(item.issue);
+          await handleOpenBranchSession(branchName, item.issue.repoPath, intent.prompt ?? undefined);
+        }
+        break;
+      }
+      case 'archive': {
```
Comment (donovan-yohan) [inline frontend/src/App.svelte:753]: **[silent-failure-hunter]** The `archive` case is an empty `break` — the user clicks "Archive" in the picker and nothing happens. No feedback, no error, no TODO comment.

Additionally, several guards in this switch silently drop actions:
- L711: `resume-session` does nothing when `existingSessionId` is falsy (indicates a bug in `resolveIntent`)
- L734: `open-branch` does nothing when `!activeWorkspace` (user gets no feedback)

Fix: Either implement archive (call `handleArchive` at L872) or show a "not yet implemented" message. Add `console.warn` for the guard fallthrough cases.

File: frontend/src/App.svelte [hunk 5]
```diff
@@ -685,17 +705,85 @@
...
+      await refreshAll();
+      sessionState.activeSessionId = session.id;
+      ui.activeRepoPath = repoPath;
+      initSessionNotification(session.id, configState.defaultNotifications);
+      closeSidebar();
+
+      if (prompt) {
+        setTimeout(() => sendPtyData(prompt + '\r'), 1500);
+      }
+    } catch (e) {
```
Comment (donovan-yohan) [inline frontend/src/App.svelte:800]: **[silent-failure-hunter]** `console.error` only — no user-facing feedback when session/worktree creation fails. Same pattern at L658 and L703. The user clicks a picker item, the picker closes, and nothing visibly happens.

**[code-simplifier]** This function largely duplicates `handleOpenPrBranch` (L663-706) — both create a worktree, create a session, refresh, set active, and optionally send a prompt. Consider sharing a common core.

File: frontend/src/App.svelte [hunk 6]
```diff
@@ -685,17 +705,85 @@
...
+        branchName: wt.branchName,
+      });
+      await refreshAll();
+      sessionState.activeSessionId = session.id;
+      ui.activeRepoPath = repoPath;
+      initSessionNotification(session.id, configState.defaultNotifications);
+      closeSidebar();
+
+      if (prompt) {
+        setTimeout(() => sendPtyData(prompt + '\r'), 1500);
```
Comment (donovan-yohan) [inline frontend/src/App.svelte:798]: **[silent-failure-hunter]** `setTimeout(1500)` is a race condition for prompt delivery. If the WebSocket isn't connected within 1500ms (slow network, server under load), `sendPtyData` silently fails and the prompt is lost. Same pattern exists at L699.

Consider using an event-driven approach (wait for terminal ready signal) or at minimum extract the magic number to a named constant.

File: frontend/src/lib/session-intent.ts [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+  _worktrees: WorktreeInfo[],
+): SessionIntent[] {
+  const intents: SessionIntent[] = [];
+
+  // Check for existing session on this PR's branch
+  const existingSession = sessions.find(s => s.branchName === pr.headRefName);
+
+  // Build PrStateInput from PullRequest data
+  const prStateInput: PrStateInput = {
+    commitsAhead: 1,
```
Comment (donovan-yohan) [inline frontend/src/lib/session-intent.ts]: **[code-simplifier]** This `PullRequest → PrStateInput` mapping is duplicated 4 times across the codebase:
1. Here (session-intent.ts:63-73)
2. OrgDashboard.svelte:80-92
3. RepoDashboard.svelte:53-65
4. App.svelte:776-786

Extract a shared `buildPrStateInput(pr: PullRequest): PrStateInput` helper in `pr-state.ts` to eliminate ~40 duplicate lines and ensure consistency when the mapping logic changes.

File: frontend/src/lib/session-intent.ts [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+    prState: pr.isDraft ? 'DRAFT' : pr.state,
+    ciPassing: pr.ciStatus === 'SUCCESS' ? 1 : 0,
+    ciFailing: (pr.ciStatus === 'FAILURE' || pr.ciStatus === 'ERROR') ? 1 : 0,
+    ciPending: pr.ciStatus === 'PENDING' ? 1 : 0,
+    ciTotal: pr.ciStatus ? 1 : 0,
+    mergeable: pr.mergeable,
+    unresolvedCommentCount: pr.reviewDecision === 'CHANGES_REQUESTED' ? 1 : 0,
```
Comment (donovan-yohan) [inline frontend/src/lib/session-intent.ts]: **[type-design-analyzer]** `unresolvedCommentCount` is derived from `reviewDecision === 'CHANGES_REQUESTED' ? 1 : 0` — a lossy approximation. A PR with 7 unresolved threads will show "Resolve Comments (1)". The `PrInfo` type in `types.ts:216` has `unresolvedCommentCount`, suggesting a data gap between the two PR representations.

File: frontend/src/lib/session-intent.ts [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+  return [openIntent];
+}
+
+function resolveIssueIntent(
+  issue: GitHubIssue,
+  sessions: SessionSummary[],
+): SessionIntent[] {
+  // Check if a session already exists for a branch matching this issue
+  const issueBranchPattern = `issue-${issue.number}`;
+  const existingSession = sessions.find(s => s.branchName.includes(issueBranchPattern));
```
Comment (donovan-yohan) [inline frontend/src/lib/session-intent.ts]: **[type-design-analyzer]** Substring match `s.branchName.includes('issue-${number}')` will incorrectly match issue 12 against a branch named `issue-123-foo` since `"issue-12"` is a substring of `"issue-123"`. Use a regex with a word boundary or exact segment match (e.g., `-issue-45-` or ending with `-45`).

**[pr-test-analyzer]** This entire issue-with-existing-session resume path is untested. The fuzzy match bug would be caught by a test.

File: frontend/src/lib/session-intent.ts [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+
+function buildIssuePrompt(issue: GitHubIssue): string {
+  const labels = issue.labels.map(l => l.name).join(', ');
+  return `Work on issue #${issue.number}: ${issue.title}\n\nLabels: ${labels}`;
+}
+
+/**
+ * Derive a branch name from a GitHub issue.
+ * Format: {type}/issue-{number}-{slug}
+ * Type is derived from labels: bug → fix, enhancement/feature → feat, default → feat
```
Comment (donovan-yohan) [inline frontend/src/lib/session-intent.ts]: **[comment-analyzer]** JSDoc says `Type is derived from labels: bug → fix, enhancement/feature → feat, default → feat` but the code (L205) only checks for `'bug'` — there is no explicit `enhancement` or `feature` check. The comment implies these labels are specially recognized when they just fall through to the default.

Fix: Change to `Type is derived from labels: bug → fix, all others → feat`

**[pr-test-analyzer]** `issueToBranchName` is exported but has zero test coverage. It has non-trivial regex/string logic and a conditional prefix that needs tests for: bug labels, special characters, empty titles, long titles.

File: frontend/src/components/OpenPicker.svelte [hunk 11]
```diff
@@ -0,0 +1,489 @@
...
+
+      <!-- Results -->
+      <div class="picker-results" role="listbox">
+        {#if flatRows.length === 0}
+          <div class="picker-empty">{emptyMessage}</div>
+        {:else}
+          {#each groups as group}
+            <div class="picker-category">{group.label}</div>
+            {#each group.rows as row}
+              {@const globalIndex = flatRows.indexOf(row)}
```
Comment (donovan-yohan) [inline frontend/src/components/OpenPicker.svelte]: **[code-simplifier]** `flatRows.indexOf(row)` is O(n) per row, making the render loop O(n²) total. For typical picker sizes (10-50 items) this is fine functionally, but a precomputed `Map<PickerRow, number>` or cumulative offset per group would be cleaner.

File: frontend/src/components/OpenPicker.svelte [hunk 12]
```diff
@@ -0,0 +1,489 @@
...
+
+  let activeTab = $state<TabId>('all');
+  let query = $state('');
+  let focusedIndex = $state(0);
+  let inputWrapperEl = $state<HTMLDivElement | undefined>(undefined);
+
+  const queryClient = useQueryClient();
+
+  // Data sources — read from TanStack Query cache + fetch branches on demand
+  let cachedPrs = $derived<PullRequest[]>(
```
Comment (donovan-yohan) [inline frontend/src/components/OpenPicker.svelte]: **[silent-failure-hunter]** `getQueryData` returns `undefined` when the cache is cold (user opens picker before visiting org dashboard). The `?? []` fallback makes the PR and issue tabs show "no open pull requests" when data simply hasn't been fetched yet. No loading state for these tabs.

Consider converting to `createQuery` calls (like `branchQuery`) with loading indicators, or distinguish between "not fetched" and "fetched but empty."

File: frontend/src/components/OpenPicker.svelte [hunk 13]
```diff
@@ -0,0 +1,489 @@
...
+  );
+
+  let filteredIssues = $derived(
+    q ? cachedIssues.filter(i =>
+      i.title.toLowerCase().includes(q) ||
+      String(i.number).includes(q)
+    ) : cachedIssues
+  );
+
+  // Group PRs by role
```
Comment (donovan-yohan) [inline frontend/src/components/OpenPicker.svelte]: **[comment-analyzer]** Comment says `// Group PRs by role` but `authorPrs` (L82) includes `pr.state !== 'OPEN'` — meaning closed/merged PRs from any role (including reviewer) end up in `authorPrs`. This is grouping by role + open-state, not purely by role.

Fix: `// Split PRs: open review requests vs everything else (authored + closed/merged)`

File: frontend/src/lib/session-intent.ts [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+  | 'fix-errors'
+  | 'resolve-comments'
+  | 'merge-pr'
+  | 'create-pr'
+  | 'open-branch'
+  | 'start-from-issue'
+  | 'resume-session'
+  | 'archive';
+
+export interface SessionIntent {
```
Comment (donovan-yohan) [inline frontend/src/lib/session-intent.ts:21]: **[type-design-analyzer]** `SessionIntent` uses optional fields (`existingSessionId?`, `existingWorktreePath?`) whose presence is coupled to `type` but not expressed in the type system. A `resume-session` intent without `existingSessionId` compiles but produces a broken UI path (App.svelte:711 checks it and does nothing if undefined).

Consider a discriminated union:
```typescript
type SessionIntent = ResumeIntent | ActionIntent;
interface ResumeIntent { type: 'resume-session'; existingSessionId: string; /* ... */ }
interface ActionIntent { type: Exclude<SessionIntentType, 'resume-session'>; /* ... */ }
```

File: frontend/src/components/PickerResultRow.svelte [hunk 15]
```diff
@@ -0,0 +1,162 @@
...
+      </TuiButton>
+    {/each}
+    {#if primary}
+      <TuiButton
+        variant={colorToVariant(primary.color)}
+        size="sm"
+        onclick={(e) => { e.stopPropagation(); onSelectIntent(primary); }}
+      >
+        {primary.label}
+      </TuiButton>
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/components/PickerResultRow.svelte]: Fixed — added `primary.label` guard so empty-label intents no longer render a blank button.

File: frontend/src/components/OpenPicker.svelte [hunk 16]
```diff
@@ -0,0 +1,489 @@
...
+      <!-- Results -->
+      <div class="picker-results" role="listbox">
+        {#if flatRows.length === 0}
+          <div class="picker-empty">{emptyMessage}</div>
+        {:else}
+          {#each groups as group}
+            <div class="picker-category">{group.label}</div>
+            {#each group.rows as row}
+              {@const globalIndex = flatRows.indexOf(row)}
+              <PickerResultRow
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/components/OpenPicker.svelte:319]: Fixed — precomputed `rowIndexMap` via `new Map(flatRows.map((row, i) => [row, i]))`, replacing O(n²) indexOf.

File: frontend/src/App.svelte [hunk 17]
```diff
@@ -685,17 +705,85 @@
...
+        break;
+      }
+      case 'review-pr':
+      case 'fix-conflicts':
+      case 'fix-errors':
+      case 'resolve-comments':
+      case 'create-pr': {
+        if (item.kind === 'pr') {
+          handleOpenPrBranch(item.pr, intent.prompt ?? undefined);
+        }
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/App.svelte:731]: Fixed — `fix-conflicts` intent now routes to `handleFixConflicts(item.pr)` which preserves session reuse and custom prompt support.

File: frontend/src/App.svelte [hunk 18]
```diff
@@ -685,17 +705,85 @@
...
   }
 
+  async function handlePickerIntent(intent: SessionIntent, item: PickerItem) {
+    switch (intent.type) {
+      case 'resume-session': {
+        if (intent.existingSessionId) {
+          sessionState.activeSessionId = intent.existingSessionId;
+          closeSidebar();
+        }
+        break;
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/App.svelte:723]: Fixed — `resume-session` now calls `navigateToSession()` which sets `activeRepoPath`, calls `handleUserViewed`, and closes the sidebar.

File: frontend/src/lib/session-intent.ts [hunk 19]
```diff
@@ -0,0 +1,213 @@
...
+function mapPrActionToIntent(actionType: string): SessionIntentType {
+  switch (actionType) {
+    case 'review-pr': return 'review-pr';
+    case 'merge-pr': return 'merge-pr';
+    case 'fix-conflicts': return 'fix-conflicts';
+    case 'fix-errors': return 'fix-errors';
+    case 'resolve-comments': return 'resolve-comments';
+    case 'create-pr': return 'create-pr';
+    case 'archive-merged':
+    case 'archive-closed': return 'archive';
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts:113]: Fixed — parameter typed as `PrActionType` with all 11 variants handled explicitly (`ready-for-review`, `checks-running`, `none` → `open-branch`). No more default fallthrough.

File: frontend/src/App.svelte [hunk 20]
```diff
@@ -685,17 +705,85 @@
...
+      const wt = await createWorktree(repoPath, branchName);
+      const session = await createSession({
+        repoPath,
+        worktreePath: wt.worktreePath,
+        type: 'agent',
+        branchName: wt.branchName,
+      });
+      await refreshAll();
+      sessionState.activeSessionId = session.id;
+      ui.activeRepoPath = repoPath;
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/App.svelte:792]: Fixed — `handleOpenBranchSession` now checks for existing sessions and worktrees before calling `createWorktree`, mirroring the pattern in `handleOpenPrBranch`.

File: frontend/src/components/OrgDashboard.svelte [hunk 21]
```diff
@@ -77,16 +77,17 @@
...
-      ciTotal: 0,
+      ciPassing: pr.ciStatus === 'SUCCESS' ? 1 : 0,
+      ciFailing: (pr.ciStatus === 'FAILURE' || pr.ciStatus === 'ERROR') ? 1 : 0,
+      ciPending: pr.ciStatus === 'PENDING' ? 1 : 0,
+      ciTotal: pr.ciStatus ? 1 : 0,
       mergeable: (pr.mergeable as 'MERGEABLE' | 'CONFLICTING' | 'UNKNOWN' | null) ?? null,
-      unresolvedCommentCount: 0,
+      unresolvedCommentCount: pr.reviewDecision === 'CHANGES_REQUESTED' ? 1 : 0,
+      role: pr.role,
     });
```
[the remark this answers is not in the document]
Comment (donovan-yohan) [reply frontend/src/components/OrgDashboard.svelte]: Fixed — OrgDashboard maps `merge-pr`, `archive-merged`, and `archive-closed` to neutral "Open" label since the button navigates to the workspace, not performs the action.

File: frontend/src/App.svelte (hunk 1 above)
Comment (donovan-yohan) [reply frontend/src/App.svelte]: Fixed in 899efe9 — added existence checks for sessions/worktrees in `handleOpenBranchSession`.

File: frontend/src/App.svelte (hunk 2 above)
Comment (donovan-yohan) [reply frontend/src/App.svelte]: Fixed — uses `item.repoPath` directly, removed `activeWorkspace` guard.

File: frontend/src/lib/session-intent.ts (hunk 3 above)
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts]: Fixed — typed as `PrActionType`, all 11 variants explicit.

File: frontend/src/App.svelte (hunk 4 above)
Comment (donovan-yohan) [reply frontend/src/App.svelte:753]: Fixed — archive wired to `handleArchive()`, resume-session warns on missing ID.

File: frontend/src/App.svelte (hunk 5 above)
Comment (donovan-yohan) [reply frontend/src/App.svelte:800]: Addressed — function now reuses existing sessions/worktrees. console.error pattern is pre-existing; added TODO for setTimeout race.

File: frontend/src/App.svelte (hunk 6 above)
Comment (donovan-yohan) [reply frontend/src/App.svelte:798]: Added TODO comment. Pre-existing pattern (also at L699) — fixing needs event-driven terminal-ready signal.

File: frontend/src/lib/session-intent.ts (hunk 7 above)
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts]: Fixed — extracted `buildPrStateInput()` in pr-state.ts, all 4 call sites now use it.

File: frontend/src/lib/session-intent.ts (hunk 9 above)
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts]: Fixed — regex now uses `issue-${number}(?:-|$)` to avoid false positives. Added tests for both the false-positive and the correct match.

File: frontend/src/lib/session-intent.ts (hunk 10 above)
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts]: Fixed — JSDoc updated to "bug → fix, all others → feat". Added tests for issueToBranchName covering bug label, special chars, truncation.

File: frontend/src/components/OpenPicker.svelte (hunk 11 above)
Comment (donovan-yohan) [reply frontend/src/components/OpenPicker.svelte]: Fixed — precomputed `rowIndexMap`.

File: frontend/src/components/OpenPicker.svelte (hunk 13 above)
Comment (donovan-yohan) [reply frontend/src/components/OpenPicker.svelte]: Fixed — comment updated to "Split PRs: open review requests vs everything else (authored + closed/merged)".

File: frontend/src/lib/session-intent.ts (hunk 14 above)
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts:21]: Added TODO comment documenting the discriminated union improvement for a follow-up.

File: frontend/src/lib/session-intent.ts (hunk 8 above)
Comment (donovan-yohan) [reply frontend/src/lib/session-intent.ts]: Added TODO comment. Known data gap — PullRequest type needs API-level change to carry actual count.

File: frontend/src/components/OpenPicker.svelte (hunk 12 above)
Comment (donovan-yohan) [reply frontend/src/components/OpenPicker.svelte]: Leaving open for discussion. Converting to `createQuery` would add loading indicators but also extra network requests. Current approach is intentional — the picker uses warm cache from OrgDashboard. Worth revisiting if picker becomes a primary entry point.

Comment (donovan-yohan) [reply frontend/src/components/OpenPicker.svelte]: Fixed in 1626d72 — picker now uses `createQuery` for PRs and issues. TanStack deduplicates with existing queries, so no extra requests when cache is warm. Shows "loading..." when cache is cold.
